### PR TITLE
Logs

### PR DESCRIPTION
**Pull request by peter-edge**
Redo of https://github.com/grpc/grpc-go/pull/190

### CODE REVIEW

**Comment by peter-edge** [comment]
Of note, it would be nice if I could put this right in the grpc package, but this would lead to cyclic dependencies (transport <-> grpc, for one) - maybe a bigger overall structural issue we could address?

**Comment by iamqizhao** [comment]
On Fri, May 8, 2015 at 4:25 PM, Peter Edge <notifications@github.com> wrote:

> Of note, it would be nice if I could put this right in the grpc package,
> but this would lead to cyclic dependencies (transport <-> grpc, for one) -
> maybe a bigger overall structural issue we could address?
>
It is okay to put it into a separate package.

File: log/logger.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+/*
+Package log defines logging for grpc.
+*/
+package log // import "google.golang.org/grpc/log"
```
Comment (iamqizhao) [inline log/logger.go]: probably rename to rpclog to avoid conflicting with the standard one.

**Comment by iamqizhao** [comment]
btw, having a separate transport package is intended for some proxy type
usage which does not care the upper layer semantics (in grpc package).

**Comment by peter-edge** [comment]
Go standards might actually dictate to name it log, but I agree that it's confusing. Naming it 'grpclog' is a decent compromose - rpclog doesn't have any corollaries in this package, we should stay consistent. I redid this.

**Pull request merged**